source/module_esolver/esolver_ks.cpp

```diff
@@ -623,6 +630,36 @@ void ESolver_KS<T, Device>::runner(const int istep, UnitCell& ucell)
                                  duration);
 #endif //__RAPIDJSON
 
+        // 12.5) rdmft, add by jghan 2024-04-08/2024-10-09
+        if ( PARAM.inp.ab_initio_type == "rdmft" )
```

Comment:
I guess this part can be put into `ESolver_KS_LCAO::iter_finish`, where you can directly call `rdmft_solver.run` and `rdmft_solver.update_elec`. Then the wrapper functions in esolver_ks.h can be removed.